### ISSUE TITLE
AMD 17.12.1 (or 17.12.2) + Prismatik + game = crashes

### ISSUE THREAD

**Issue by Garwinski** [psieg/Lightpack#159]
Hi, 

I have been using your fork of Lightpack since the beginning. Since the 17.12 (17.12.1 and 17.12.2) drivers from AMD, the driver seems to have difficulty with two or more programs calling on the gpu, and then crashes. With Wolfenstein II and the vulkan renderer, the game wont even start anymore when I have Prismatik running. When I do try it, I will get a black screen, and the Radeon setting host application crashes, indicating the driver crashes. Vulkan is pretty new, so some issues could be expected, as it might work totally differently from directx 9/10/11 applications in terms of calling on the gpu.

 On some other games, I will get a generic random DirectX error after some time, while at the same time the Radeon settings host application crashes, and event viewer also logs a freeze and recovery of the gpu driver.  With GTA V for example, I can get some crashes from time to time with the error: ERR_GFX_D3D_INIT. In event viewer, the system reports "Display driver amdkmdap stopped responding and has successfully recovered." at the same time the application (GTA V in this case) crashes. Especially in GTA V I got this error a lot on the same section, and could not get past this section. After closing Prismatik, I was able to pass it, and have played it for three hours straight without an error. 

Did something fundamentally change in regards to how the driver and applications like Prismatik (desktop duplication) work together in these drivers? Does it have anything to do with the creators update and the new driver model that came with it?

**Comment by psieg** [comment]
I have no idea about changes in AMD drivers honestly. But experience suggest this is the driver's fault.

**If DX hooking is off**, the only thing you can do is switch to WinAPI capturing as a workaround.

**Comment by psieg** [comment]
Any update on this? Does it work with newer versions or did you stick to WinAPI?

**Comment by Garwinski** [comment]
Damn, I am sorry. I had typed a response here, but for some reason it did not send. 

GTA V still crashes, even with Prismatik off. There seem to be some problems with drivers starting from 17.12 and GTA V in general, and general problems with these drivers for Fury cards (Fiji) are reported often as well. 

In relation to your Prismatik fork: For Wolfenstein II, I dont mind as much, as it uses a new API, and all parties involved in this game and the renderer are still orienting themselves in the new tech. I have not tried this game in a while though. Maybe with the new released Vulkan 1.1 they fixed some relevant things. I will test this.

GTA V seems to still have problems in general. Even though the crashes are far less often when I have prismatik disabled, they are still there, and are being reported by a lot of people with AMD cards since 17.12. So this is something between Rockstar and AMD. That I could pass that one section after disabling Prismatik does probably not point to a problem with Prismatik and GTA V, but more to a problem with GTA V and gpu-resource sharing between applications in general. 

With other games I have no crashes at all, and I enjoy using your Prismatik fork with everything, as it just works, I practically never have to think about it or adjust something to Prismatik for it to work. I just start op my pc, and start doing what I want, and it works. Damn, those times when playing games with Prismatik was a pain the **se, with dx hooking, or using Ambibox with Playclaw etc... 

Thanks for your effort.

**Comment by Benik3** [comment]
I have too AMD (RX580).
Prismatik works fine in all my games except games using Vulkan engine (Wolfenstein...). GTA V I don't have so I didn't test...
Now I have 18.1.1 drivers.

**Comment by psieg** [comment]
@Benik3, Are you using DX injection? Does it work without?

**Comment by Benik3** [comment]
Do you mean the settings in Prismatik?
I use windows 8 function (and OS I have windows 10)

**Comment by Garwinski** [comment]
To follow up on this: 

I updated to the newest vulkan runtimes, and reinstalled all visual c++ versions, as they sometimes can get overwritten with older versions when you install a game, or get corrupted. ( I use this program, which uninstalls them all and then reinstalls the newest versions: https://forums.mydigitallife.net/threads/repack-visual-c-redistributable-runtimes-2018-06-28.76588/ ) It is a good practice to do this from time to time, and the linked program makes it easy. I had problems with micro stutter in some games some time ago and had exhausted all options I thought (reinstalling drivers, reinstalling games, dabbling with settings, tweaking my BIOS settings etc) , but this fixed that problem, so I decided to reinstall this stuff again now.

Now I can play Wolfenstein II with Prismatik! I am almost sure it was the reinstalling of vc++ stuff, but installing the newest version of Vulkan didnt hurt either probably. 

PS I use Windows 8 desktop dupplication, DX hooking is disabled, as this has not been necessary to have Prismatik work with games since Windows 10 I believe.
